core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -440,6 +441,8 @@ class KafkaServer(
             threadNamePrefix,
             CompletableFuture.completedFuture(quorumVoters),
             QuorumConfig.parseBootstrapServers(config.quorumBootstrapServers),
+            // Endpoint information is only needed for controllers (voters). ZK brokers can never be controllers
+            Endpoints.empty(),
```

Comment:
Should we make it clear that `ZK brokers can never be [KRaft] controllers`.

Reply:
Fixed the comment.